notifications/views.py

```diff
@@ -27,5 +29,14 @@ class NotificationSettingsViewSet(
 
     def get_queryset(self):
         """Gets the QuerySet for this view"""
+
+        admin_api = get_admin_api()
+
         user = self.request.user
-        return user.notification_settings
+
+        if admin_api.is_moderator_for_any_channel(user.username):
+            return user.notification_settings
+        else:
+            return user.notification_settings.exclude(
+                notification_type=NOTIFICATION_TYPE_MODERATOR
+            )
```

Comment:
Is this part necessary, would a non-moderator not have that notificiation type?

Reply:
Yes, we do not delete the notification setting if a user is removed as a moderator

Reply:
I updated the code to remove the moderator setting when a user is removed as a moderator